yt/frontends/halo_catalog/data_structures.py

```diff
@@ -146,3 +177,363 @@ def _is_valid(self, *args, **kwargs):
+
+    def _read_particle_fields(self, fields, dobj, chunk=None):
+        if len(fields) == 0:
+            return {}, []
+        fields_to_read, fields_to_generate = self._split_fields(fields)
+        if len(fields_to_read) == 0:
+            return {}, fields_to_generate
+        fields_to_return = self.io._read_particle_selection(dobj, fields_to_read)
+        return fields_to_return, fields_to_generate
+
+    def _setup_data_io(self):
+        super(YTHaloParticleIndex, self)._setup_data_io()
+        if self.real_ds._instantiated_index is None:
+            self.real_ds.index
+
+        # inherit some things from parent index
+        for attr in ["data_files", "total_particles"]:
+            setattr(self, attr, getattr(self.real_ds.index, attr))
+
+        self._calculate_particle_index_starts()
+        self._create_halo_id_table()
+
+
+class HaloDataset(ParticleDataset):
+    """
+    Base class for dataset accessing particles from halo catalogs.
+    """
+
+    def __init__(self, ds, dataset_type):
+        self.real_ds = ds
+        for attr in [
+            "filename_template",
+            "file_count",
+            "particle_types_raw",
+            "particle_types",
+            "periodicity",
+        ]:
+            setattr(self, attr, getattr(self.real_ds, attr))
+
+        super(HaloDataset, self).__init__(self.real_ds.parameter_filename, dataset_type)
+
+    def print_key_parameters(self):
+        pass
+
+    def _set_derived_attrs(self):
+        pass
+
+    def _parse_parameter_file(self):
+        for attr in [
+            "cosmological_simulation",
+            "cosmology",
+            "current_redshift",
+            "current_time",
+            "dimensionality",
+            "domain_dimensions",
+            "domain_left_edge",
+            "domain_right_edge",
+            "domain_width",
+            "hubble_constant",
+            "omega_lambda",
+            "omega_matter",
+            "unique_identifier",
+        ]:
+            setattr(self, attr, getattr(self.real_ds, attr))
+
+    def set_code_units(self):
+        for unit in ["length", "time", "mass", "velocity", "magnetic", "temperature"]:
+            my_unit = "%s_unit" % unit
+            setattr(self, my_unit, getattr(self.real_ds, my_unit, None))
+        self.unit_registry = self.real_ds.unit_registry
+
+    def __repr__(self):
+        return "%s" % self.real_ds
+
+    def _setup_classes(self):
+        self.objects = []
+
+
+class YTHaloDataset(HaloDataset):
+    """
+    Dataset used for accessing member particles from YTHaloCatalogDatasets.
+    """
+
+    _index_class = YTHaloParticleIndex
+    _file_class = YTHaloCatalogFile
+    _field_info_class = YTHaloCatalogHaloFieldInfo
+
+    def __init__(self, ds, dataset_type="ythalo"):
+        super(YTHaloDataset, self).__init__(ds, dataset_type)
+
+    def _set_code_unit_attributes(self):
+        pass
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        return False
```

Comment:
same as above. If it's really not meant not to be instanciated with yt.load, please state so (we currently have at least one dataset class that should be loadable but isn't because the validator was never completed)